server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -142,7 +145,28 @@ public static void download(Repository repository, ShardId shardId, ThreadPool t
         RemoteFsTranslog.download(translogTransferManager, location, logger);
     }
 
-    public static void download(TranslogTransferManager translogTransferManager, Path location, Logger logger) throws IOException {
+    static void download(TranslogTransferManager translogTransferManager, Path location, Logger logger) throws IOException {
+        /*
+        In Primary to Primary relocation , there can be concurrent upload and download of translog.
+        While translog files are getting downloaded by new primary, it might hence be deleted by the primary
+        Hence we retry if tlog/ckp files are not found .
+
+        This doesn't happen in last download , where it is ensured that older primary has stopped modifying tlog data.
+         */
+        IOException ex = null;
+        for (int i = 0; i <= DOWNLOAD_RETRIES; i++) {
+            try {
+                downloadOnce(translogTransferManager, location, logger);
+                return;
+            } catch (FileNotFoundException | NoSuchFileException e) {
+                // continue till download retries
+                ex = e;
+            }
+        }
+        throw ex;
```

Comment:
Should we use the constantBackoff with maxRetries method present in `BackoffPolicy` class?

Reply:
`BackoffPolicy` and `RetryableAction` are made for transport  requests  . For our usecase , this will only complicate the code. .